### PR TITLE
Evolution Scene and other data dism

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/evolution_scene.h [hunk 1]
```diff
@@ -8,6 +8,6 @@ void EvolutionScene(struct Pokemon* mon, u16 speciesToEvolve, u8, u8 partyID);
 void TradeEvolutionScene(struct Pokemon* mon, u16 speciesToEvolve, u8 preEvoSpriteID, u8 partyID);
 
 extern void (*gCB2_AfterEvolution)(void);
```
Comment (jiangzhengwenjz) [inline include/evolution_scene.h]: This line should be above function DECLs

File: include/evolution_scene.h [hunk 2]
```diff
@@ -8,6 +8,6 @@ void EvolutionScene(struct Pokemon* mon, u16 speciesToEvolve, u8, u8 partyID);
 void TradeEvolutionScene(struct Pokemon* mon, u16 speciesToEvolve, u8 preEvoSpriteID, u8 partyID);
```
Comment (jiangzhengwenjz) [inline include/evolution_scene.h]: Can you update ID to Id in this file?

File: src/evolution_scene.c [hunk 3]
```diff
@@ -0,0 +1,1566 @@
...
+// const data
+static const u16 sUnrefPal_83F6C90[] = INCBIN_U16("graphics/evolution_scene/unknown_5B4114.gbapal");
+static const u32 sMovingBackgroundTiles[] = INCBIN_U32("graphics/evolution_scene/bg.4bpp.lz");
+static const u32 sMovingBackgroundMap1[] = INCBIN_U32("graphics/evolution_scene/bg.bin.lz");
+static const u32 sMovingBackgroundMap2[] = INCBIN_U32("graphics/evolution_scene/bg2.bin.lz");
+static const u16 sBlackPalette[] = INCBIN_U16("graphics/evolution_scene/gray_transition_intro.gbapal");
+static const u16 unref_83F7D80[] = INCBIN_U16("graphics/evolution_scene/unref_83F7D80.bin");
+static const u16 sMovingBgPals[] = INCBIN_U16("graphics/evolution_scene/transition.gbapal");
+
+static const u8 Text_ShedinjaJapaneseName[] = _("ヌケニン");
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: `sText_ShedinjaJapaneseName`

File: src/evolution_scene.c [hunk 4]
```diff
@@ -0,0 +1,1566 @@
...
+            LinkTradeDrawWindow();
+        }
+        break;
+    case 6:
+        if (gWirelessCommType)
+        {
+            LoadWirelessStatusIndicatorSpriteGfx();
+            CreateWirelessStatusIndicatorSprite(0, 0);
+        }
+        BlendPalettes(-1,0x10, 0);
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: 1) first arg should be `0xFFFFFFFF`
2) `RGB_BLACK`

File: src/evolution_scene.c [hunk 5]
```diff
@@ -0,0 +1,1566 @@
...
+    gBattle_BG0_X = 0;
+    gBattle_BG0_Y = 0;
+    gBattle_BG1_X = 0;
+    gBattle_BG1_Y = 0;
+    gBattle_BG2_X = 0;
+    gBattle_BG2_Y = 0;
+    gBattle_BG3_X = 256;
+    gBattle_BG3_Y = 0;
+
+    gTextFlags.useAlternateDownArrow = 1;
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: `TRUE`

File: src/evolution_scene.c [hunk 6]
```diff
@@ -0,0 +1,1566 @@
...
+            }
+        }
+        break;
+    case 14:
+        if (!IsTextPrinterActive(0))
+        {
+            DestroyTask(taskId);
+            Free(sEvoStructPtr);
+            sEvoStructPtr = NULL;
+            gTextFlags.useAlternateDownArrow = 0;
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: `FALSE`

File: src/evolution_scene.c [hunk 7]
```diff
@@ -0,0 +1,1566 @@
...
+                DrawTextOnTradeWindow(0, gText_CommunicationStandby5, 1);
+                gTasks[taskId].tState++;
+            }
+        }
+        break;
+    case 14:
+        if (!IsTextPrinterActive(0))
+        {
+            DestroyTask(taskId);
+            Free(sEvoStructPtr);
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: use `FREE_AND_SET_NULL` macro

File: src/evolution_scene.c [hunk 8]
```diff
@@ -0,0 +1,1566 @@
...
+                SetMonData(shedinja, MON_DATA_NICKNAME, Text_ShedinjaJapaneseName);
+    }
+}
+
+static void Task_EvolutionScene(u8 taskId)
+{
+    u32 var;
+    struct Pokemon* mon = &gPlayerParty[gTasks[taskId].tPartyId];
+
+    // National dex check
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: This comment has no use here as the function is named

File: src/evolution_scene.c [hunk 9]
```diff
@@ -0,0 +1,1566 @@
...
+                StopMapMusic();
+                Overworld_PlaySpecialMapMusic();
+            }
+            if (!gTasks[taskId].tEvoWasStopped)
+                CreateShedinja(gTasks[taskId].tPreEvoSpecies, mon);
+
+            DestroyTask(taskId);
+            FreeMonSpritesGfx();
+            Free(sEvoStructPtr);
+            sEvoStructPtr = NULL;
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: use `FREE_AND_SET_NULL` macro

File: src/evolution_scene.c [hunk 10]
```diff
@@ -0,0 +1,1566 @@
...
+        break;
+    }
+}
+
+static void Task_TradeEvolutionScene(u8 taskId)
+{
+    u32 var = 0;
+    struct Pokemon* mon = &gPlayerParty[gTasks[taskId].tPartyId];
+
+    // National dex check
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: same here

File: src/evolution_scene.c [hunk 11]
```diff
@@ -0,0 +1,1566 @@
...
+        }
+        break;
+    case 4:
+        if (!gPaletteFade.active)
+        {
+            InitMovingBackgroundTask(TRUE);
+            var = gSprites[sEvoStructPtr->preEvoSpriteId].oam.paletteNum + 16;
+            sEvoGraphicsTaskId = LaunchTask_PreEvoSparklesSet1(var);
+            gTasks[taskId].tState++;
+            SetGpuReg(REG_OFFSET_BG3CNT, 0x603);
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: use macros

File: src/evolution_scene.c [hunk 12]
```diff
@@ -0,0 +1,1566 @@
...
+            PlayCry1(gTasks[taskId].tPreEvoSpecies, 0);
+            gTasks[taskId].tState++;
+        }
+        break;
+    case 17:
+        if (IsCryFinished())
+        {
+            StringExpandPlaceholders(gStringVar4, gText_EllipsisQuestionMark);
+            DrawTextOnTradeWindow(0, gStringVar4, 1);
+            gTasks[taskId].tEvoWasStopped = 1;
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: `TRUE`

File: src/evolution_scene.c [hunk 13]
```diff
@@ -0,0 +1,1566 @@
...
+}
+
+void IsMovingBackgroundTaskRunning(void) // unused
+{
+    u8 taskId = FindTaskIdByFunc(Task_MovingBackgroundPalettes);
+
+    if (taskId != 0xFF)
+        gTasks[taskId].data[6] = 1;
+
+    FillPalette(0, 0xA0, 0x20);
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c]: `RGB_BLACK`

File: src/evolution_scene.c (hunk 7 above)
Comment (PikalaxALT) [reply src/evolution_scene.c]: I'm neither using nor enforcing that macro.

**Pull request merged**